.github/workflows/build-storybook-main.yml

```diff
@@ -0,0 +1,45 @@
+name: Build Storybook Main
+
+on:
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      PUBLISH_STORYBOOK_TOKEN:
+        required: true
+
+jobs:
+  build-deploy-storybook:
+    name: Build and Deploy Storybook
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    environment: github-pages
+    defaults:
+      run:
+        working-directory: apps/storybook-react
+    steps:
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
+        with:
+          is-high-risk-environment: false
+          cache-node-modules: true
+
+      - name: Build required packages
+        working-directory: .
+        run: |
+          yarn workspace @metamask/design-tokens build
+          yarn workspace @metamask/design-system-tailwind-preset build
+          yarn workspace @metamask/design-system-react build
+
+      - name: Build Storybook
+        run: yarn build-storybook
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@de7ea6f8efb354206b205ef54722213d99067935
```

Comment:
This action is used by a [few MM repos](https://github.com/search?q=org%3AMetaMask%20peaceiris%2Factions-gh-pages%40de7ea6f8efb354206b205ef54722213d99067935&type=code) to publish to github pages but open to using something else if it exists

Reply:
I believe we commonly use this action in other repos because the official action doesn't support deploying to a subdirectory. This is common for libraries because we want to have a recorded version of API docs for each historical version, so we deploy to e.g. `1.2.3/` within `gh-pages` instead of the root directory. In this case however we seem to be using the action in a different way so perhaps using an unofficial action isn't necessary.

Reply:
Thanks for the suggestion and context both! I'll update with the official action here

Reply:
Updated to align with [actions/upload-pages-artifact@v3](https://github.com/actions/upload-pages-artifact?) and  [actions/deploy-pages@v4](https://github.com/actions/deploy-pages?tab=readme-ov-file) here https://github.com/MetaMask/metamask-design-system/pull/610/commits/4cc94306a023d83cf05cb0bec676628e94fd7db8